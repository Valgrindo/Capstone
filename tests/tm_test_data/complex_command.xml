<!-- This test is a complex command command with multiple rolegroups and OR options -->
<commands>

    <command name="WIN-MOVE">
        <component indicator="I1, I2" type="T1, T2">
            <!-- First primary rolegroup -->
            <rolegroup>

                <!-- Role 1.1 -->
                <role name="R1">
                    <component id="ID1" indicator="I2" type="T3, T4, T5" word="W1, W2">
                        <role name="R1.1">
                            <component indicator="I3, I4" type="T6" word="W3, W4"/>
                            <component indicator="I5, I6" type="T7" word="W5, W6"/>
                            <component indicator="I7, I8" type="T8" word="W7, W8"/>
                        </role>
                        <role name="R1.2">
                            <component indicator="I9, I10" type="T9, T10" word="W9"/>
                        </role>
                    </component>
                </role>

                <!-- Role 1.2 -->
                <role name="N2">
                    <component id="ID2" indicator="I11" type="T11" word="W10, W11">
                        <role name="R2.1">
                            <component indicator="I12, I13" type="T12, T13" word="W12, W13, W14"/>
                        </role>
                    </component>
                    <component indicator="I12, I13" type="T12, T13" word="W12, W13, W14"/>
                </role>
            </rolegroup>

            <!-- Second primary <rolegroup> -->
            <rolegroup>
                <role name="R4">
                    <component indicator="I3" type="T3" word="W3"/>
                </role>
            </rolegroup>
        </component>
        <component indicator="I3" type="T3" word="W3"/>
    </command>

    <!-- Second command-->
    <command name="PRINT-TABLE">
        <component indicator="I4" type="T4" word="W4"/>
    </command>
</commands>