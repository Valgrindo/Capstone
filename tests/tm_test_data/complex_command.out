<command WIN-MOVE>
<component I1, I2 T1, T2>
|  <rolegroup>
|  |  <role R1>
|  |  |  <component I2 T3, T4, T5 W1, W2>
|  |  |  |  <rolegroup>
|  |  |  |  |  <role R1.1>
|  |  |  |  |  |  <component I3, I4 T6 W3, W4>
|  |  |  |  |  |  <component I5, I6 T7 W5, W6>
|  |  |  |  |  |  <component I7, I8 T8 W7, W8>
|  |  |  |  |  <role R1.2>
|  |  |  |  |  |  <component I9, I10 T9, T10 W9>
|  |  |  |  </rolegroup>
|  |  <role N2>
|  |  |  <component I11 T11 W10, W11>
|  |  |  |  <rolegroup>
|  |  |  |  |  <role R2.1>
|  |  |  |  |  |  <component I12, I13 T12, T13 W12, W13, W14>
|  |  |  |  </rolegroup>
|  |  |  <component I12, I13 T12, T13 W12, W13, W14>
|  </rolegroup>
|  <rolegroup>
|  |  <role R4>
|  |  |  <component I3 T3 W3>
|  </rolegroup>
<component I3 T3 W3>
</command><command PRINT-TABLE>
<component I4 T4 W4>
</command>